pkg/infrastructure/attribute_manager.go

```diff
@@ -32,14 +34,37 @@ const ValidationPath string = "/tmp/taxonomy/infraattributes.json#/definitions/I
 type AttributeManager struct {
 	Log            zerolog.Logger
 	Infrastructure infraattributes.Infrastructure
+	Mux            *sync.RWMutex
 }
 
 func NewAttributeManager() (*AttributeManager, error) {
 	content, err := readInfrastructure()
 	if err != nil {
 		return nil, err
 	}
-	return &AttributeManager{Log: logging.LogInit(logging.CONTROLLER, "FybrikApplication"), Infrastructure: content}, nil
+	return &AttributeManager{
+		Log:            logging.LogInit(logging.CONTROLLER, "FybrikApplication"),
+		Infrastructure: content,
+		Mux:            &sync.RWMutex{},
+	}, nil
+}
+
+func (m *AttributeManager) OnError(err error) {
+	m.Log.Error().Err(err).Msg("Error reading infrastructure attributes")
+}
+
+func (m *AttributeManager) GetOptions() monitor.FileMonitorOptions {
```

Comment:
Please add some documentation about what this does.  To what options does it refer?

Reply:
Done